### PR TITLE
Add codelist to dimensions when posted to the dataset-api

### PR DESCRIPTION
**Pull request by Matt-Guest**
### What
Additional information is required by the dataset-api to store a dimension. 

### How to review

Run tests, review code, ....

### Who can review
anyone

### CODE REVIEW

File: codelists/request.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	Name string `json:"name"`
+}
+
+//go:generate moq -out testcodelist/importclient.go -pkg testcodelist . ImportClient
+// ImportClient
+type ImportClient interface {
+	Get(path string) (*http.Response, error)
+}
+
+// GetFromInstance
```
Comment (eldeal) [inline codelists/request.go]: godoc should contain a comment

File: dimension/extraction.go [hunk 2]
```diff
@@ -37,7 +42,8 @@ func (e *MissingDimensionValues) Error() string {
 }
 
 // New returns a new Extract object for a given instance
-func New(dimensions map[string]string, dimensionColumnOffset int, headerRow []string, datasetAPIURL string, datasetAPIAuthToken string, instanceID string, line []string, maxRetries int, timeColumn int) *Extract {
+func New(dimensions map[string]string, dimensionColumnOffset int, headerRow []string, datasetAPIURL string, datasetAPIAuthToken string,
```
Comment (eldeal) [inline dimension/extraction.go]: Dont need a constructor to set exported fields on an exported struct with no logic

File: dimension/request.go [hunk 3]
```diff
@@ -6,35 +6,50 @@ import (
...
+func (request *Request) Post(httpClient *http.Client) error {
 	// TODO Instead off backing off by bumping the sleep by 10 seconds per failed
 	// request, we may want an exponential backoff
 	time.Sleep(time.Duration(request.Attempt-1) * 10 * time.Second)
 
-	path := request.DatasetAPIURL + "/instances/" + request.InstanceID + "/dimensions/" + request.Dimension + "/options/" + request.DimensionValue
+	option, err := json.Marshal(DimensionOption{Name: request.DimensionID, Value: request.Value, CodeList: request.CodeList, Code: request.Code})
+
+	path := fmt.Sprintf("%s/instances/%s/dimensions", request.DatasetAPIURL, request.InstanceID)
+	fmt.Printf("path : %s\n", path)
```
Comment (eldeal) [inline dimension/request.go]: this should probably use the logger and provide more useful context about whats happening (or be removed)

File: codelists/request.go (hunk 1 above)
Comment (Matt-Guest) [reply codelists/request.go]: Added godocs

File: dimension/extraction.go (hunk 2 above)
Comment (Matt-Guest) [reply dimension/extraction.go]: I removed this function

File: dimension/request.go (hunk 3 above)
Comment (Matt-Guest) [reply dimension/request.go]: Sorry that some debug message I left in

**Pull request merged**